docs/en/sdk-manual/web.md

```diff
@@ -0,0 +1,236 @@
+
+## Integrate the SDK
+
+### Include SDK
+
+```bash
+npm install @aws/clickstream-web
+```
+
+### Initialize the SDK
+
+You need to configure the SDK with default information before using it. Copy your configuration code from your clickstream solution control plane, the configuration code should look like as follows. You can also manually add this code snippet and replace the values of appId and endpoint after you registered app to a data pipeline in the Clickstream Analytics solution console.
+
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-web';
+
+ClickstreamAnalytics.init({
+   appId: "your appId",
+   endpoint: "https://example.com/collect",
+});
+```
+
+Your `appId` and `endpoint` are already set up in it.
+
+### Start using
+
+#### Record event
+
+Add the following code where you need to record event.
+
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-web';
+
+ClickstreamAnalytics.record({ name: 'albumVisit' });
+ClickstreamAnalytics.record({
+  name: 'buttonClick',
+  attributes: { _channel: 'SMS', Successful: true }
+});
+```
+For more usage refer to [GitHub start using](https://github.com/awslabs/clickstream-web#start-using)
+
+## Data format definition
+
+### Data types
+
+Clickstream Web SDK supports the following data types:
+
+| Data type | range                       | sample                |
+|-----------| --------------------------- | --------------------- |
+| number    | 5e-324~1.79e+308            | 12, 26854775808, 3.14 |
+| boolean   | true、false                  | true                  |
+| string    | max support 1024 characters | "clickstream"         |
+
+### Naming rules
+
+1. The event name and attribute name cannot start with a number, and only contains: uppercase and lowercase letters, numbers, underscores, if the event name is invalid, the SDK will record `_clickstream_error` event, if the attribute or user attribute name is invalid, the attribute will be discarded and also record `_clickstream_error` event.
+
+2. Do not use `_` as prefix to naming event name and attribute name, `_` is the preset from Clickstream Analytics.
+
+3. The event name and attribute name are in case-sensitive, so the event `Add_to_cart` and `add_to_cart` will be recognized as two different event.
+
+### Event and attribute limitation
+
+In order to improve the efficiency of querying and analysis, we apply limits to event data as follows:
+
+| Name                            | Suggestion            | Hard limit            | strategy                           | error code   |
+|---------------------------------|-----------------------|-----------------------|------------------------------------|--------------|
+| Event name invalid              | --                    | --                    | record `_clickstream_error`event   | 1001         |
+| Length of event name            | under 25 character    | 50 character          | discard, record error event        | 1002         |
```

Comment:
character -> characters 

same for below occurrences

Reply:
fixed.